services/managers/market_manager.py

```diff
@@ -0,0 +1,145 @@
+                    cursor = connections['market'].cursor()
+                    cursor.execute(marketManager.SQL_ADD_USER, [username_clean, username_clean, email, email, salt,
+                                                                hash, characterid, charactername])
+                    return username_clean, plain_password
+                except:
+                    logger.debug("Unsuccessful attempt to add market user %s" % username)
+                    return None
+            else:
+                logger.debug("Alliance market email %s already exists Updating instead" % email)
+                username_clean, password = marketManager.update_user_info(username)
+                return username_clean, password
+        else:
+            logger.debug("Alliance market username %s already exists Updating instead" % username)
+            username_clean, password = marketManager.update_user_info(username)
+            return username_clean, password
+
+    @staticmethod
+    def disable_user(username):
+        logger.debug("Disabling alliance market user %s " % username)
+        cursor = connections['market'].cursor()
+        cursor.execute(marketManager.SQL_DISABLE_USER, [username])
+        return True
+
+    @staticmethod
+    def update_custom_password(username, plain_password):
+        logger.debug("Updating alliance market user %s password" % username)
+        if marketManager.check_username(username):
+            username_clean = marketManager.__santatize_username(username)
+            hash = bcrypt.encrypt(plain_password, rounds=13)
+            hash_result = hash
+            rounds_striped = hash_result.strip('$2a$13$')
+            salt = rounds_striped[:22]
+            cursor = connections['market'].cursor()
+            cursor.execute(marketManager.SQL_UPDATE_PASSWORD, [hash, salt, username_clean])
+            return plain_password
+        else:
+            logger.error("Unable to update alliance market user %s password" % username)
+            return None
+
+    @staticmethod
+    def update_user_password(username):
+        logger.debug("Updating alliance market user %s password" % username)
+        if marketManager.check_username(username):
+            username_clean = marketManager.__santatize_username(username)
+            plain_password = marketManager.__generate_random_pass()
+            hash = bcrypt.encrypt(plain_password, rounds=13)
+            hash_result = hash
+            rounds_striped = hash_result.strip('$2a$13$')
+            salt = rounds_striped[:22]
+            cursor = connections['market'].cursor()
+            cursor.execute(marketManager.SQL_UPDATE_PASSWORD, [hash, salt, username_clean])
+            return plain_password
+        else:
+            logger.error("Unable to update alliance market user %s password" % username)
+            return None
+
+    @staticmethod
+    def update_user_info(username):

```

Comment:
what purpose does this serve? this is a password reset. use the password reset function